src/cheri_riscv_types.sail

```diff
@@ -1,28 +1,62 @@
-/* Extension for CHERI PTE checks */
-enum pte_check = {PTE_CAP_OK, PTE_LOAD_CAP_ERR, PTE_STORE_CAP_ERR}
+/*
+ * Extension for CHERI page table walker
+ *
+ * We are going to carry around a summary of the PTW walk for data-dependent
+ * (well, tag-dependent) behaviors by the instructions.  The elimination of
+ * ext_ptw values is, sadly, diffuse, across the PTW logic itself
+ * (ext_get_ptw_error) and the various instructions' analysis of load-side
+ * caveats (since the PTW does not have visibility into the data path).
+ */
 
-/* Extension for CHERI page-table-walks */
+enum ext_ptw_lcm = {
+  /* Tags flow as might be expected */
+  PTW_LCM_OK,
 
-enum ext_ptw = {PTW_CAP_OK, PTW_LOAD_CAP_ERR, PTW_STORE_CAP_ERR}
+  /* PTE settings require clearing tags */
+  PTW_LCM_CLR
+}
 
-let init_ext_ptw : ext_ptw = PTW_CAP_OK /* initial value of the PTW accumulator */
+enum ext_ptw_scm = {
+  /* Tag stores are permitted */
+  PTW_SCM_OK,
 
-/* Default accumulation of PTE check extensions into PTW accumulator */
-function ext_accum_ptw_result(ptw : ext_ptw, pte : pte_check) -> ext_ptw =
-  match (ptw, pte) {
-    (PTW_CAP_OK, PTE_CAP_OK)        => PTW_CAP_OK,
-    (PTW_CAP_OK, PTE_LOAD_CAP_ERR)  => PTW_LOAD_CAP_ERR,
-    (PTW_CAP_OK, PTE_STORE_CAP_ERR) => PTW_STORE_CAP_ERR,
+  /* Tag-asserting stores trap */
+  PTW_SCM_TRAP
+}
```

Comment:
I'd drop the M from these types and values, as SCM is now CD and LCM is going to be something similarly different?